### PR TITLE
Xapid 668

### PR DESCRIPTION
**Pull request by alexkhimich**

### CODE REVIEW

File: api.go [hunk 1]
```diff
@@ -287,18 +287,25 @@ func apiSetDeploymentResults(w http.ResponseWriter, r *http.Request) {
...
+
+
 func transmitDeploymentResultsToServer(validResults apiDeploymentResults) error {
 
 	retryIn := bundleRetryDelay
 	maxBackOff := 5 * time.Minute
 	backOffFunc := createBackoff(retryIn, maxBackOff)
 
-	uri, err := url.Parse(apiServerBaseURI.String())
+	_, err := url.Parse(apiServerBaseURI.String())
```
Comment (mdobson) [inline api.go]: Is this something that should be here? Or is this just a quick test that we don't get a bad URL?

File: api.go [hunk 2]
```diff
@@ -287,18 +287,25 @@ func apiSetDeploymentResults(w http.ResponseWriter, r *http.Request) {
...
 	if err != nil {
 		log.Errorf("unable to parse apiServerBaseURI %s: %v", apiServerBaseURI.String(), err)
 		return err
 	}
-	uri.Path = fmt.Sprintf("/clusters/%s/apids/%s/deployments", apidClusterID, apidInstanceID)
+	apiPath := fmt.Sprintf("%s/clusters/%s/apids/%s/deployments",apiServerBaseURI.String(), apidClusterID, apidInstanceID)
```
Comment (mdobson) [inline api.go]: Is this multi-return still? How does go handle multi-return functions in a string format like this?

File: api.go [hunk 3]
```diff
@@ -307,24 +314,23 @@ func transmitDeploymentResultsToServer(validResults apiDeploymentResults) error
 	}
 
 	for {
-		log.Debugf("transmitting deployment results to tracker: %s", string(resultJSON))
-		req, err := http.NewRequest("PUT", uri.String(), bytes.NewReader(resultJSON))
+		log.Debugf("transmitting deployment results to tracker by URL=%s data=%s",apiPath, string(resultJSON))
+		req, err := http.NewRequest("PUT", apiPath, bytes.NewReader(resultJSON))
```
Comment (mdobson) [inline api.go]: Is this error checked? Looks like `err` gets overwritten below.

**Review by mdobson — COMMENTED** [review]
General questions

File: api.go (hunk 1 above)
Comment (alexkhimich) [inline api.go]: Yes, testing it. Good idea before constructing url :) to prevent from FS/FTP attacks

File: api.go (hunk 2 above)
Comment (alexkhimich) [inline api.go]: Not multi return

File: api.go (hunk 3 above)
Comment (alexkhimich) [inline api.go]: fixed

File: api.go (hunk 1 above)
Comment (mdobson) [inline api.go]: Ah. I should've read right below it. I see now.

**Comment by mdobson** [comment]
👍

**Pull request merged**